homework/index.js

```diff
@@ -30,15 +15,162 @@
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+    fetch(url)
+      .then(response => response.json())
+      .then(data => {
+        data.sort((a, b) => a.name.localeCompare(b.name));
+        const contributorsURL = data.map(item => item.contributors_url);
+        const fetchedDataContributorsDefault = fetch(contributorsURL[0]).then(response =>
+          response.json(),
+        );
+        data.forEach(repository => {
+          createAndAppend('option', select, { text: repository.name });
+        });
+
+        const lowerCaseProperties = repositoryProperties
+          .map(prop => prop.toLowerCase())
+          .map(prop => prop.substring(0, prop.length - 2));
+        for (let j = 1; j < repositoryProperties.length; j++) {
+          const leftPanelContainerTableTr = createAndAppend('tr', leftPanelContainerTable);
+          createAndAppend('td', leftPanelContainerTableTr, {
+            text: repositoryProperties[j],
+          });
+          createAndAppend('td', leftPanelContainerTableTr, {
+            id: `property_${lowerCaseProperties[j]}`,
+          });
+        }
+
+        function assignLeftPanelValues(dataIndex) {
+          setLink('property_repository', dataIndex);
+          document.getElementById('property_description').innerText = dataIndex.description;
+          document.getElementById('property_forks').innerText = dataIndex.forks;
+          document.getElementById('property_updated').innerText = new Date(dataIndex.updated_at);
+        }
+
+        assignLeftPanelValues(data[0]);
+
+        function createRightPanel() {
+          const rigthColumnListDefault = createAndAppend('ul', rightPanel, {
+            id: 'right_column_list',
+          });
+          fetchedDataContributorsDefault.then(data2 => {
+            data2.forEach(contributor => {
+              const rightColumnListElems = createAndAppend('li', rigthColumnListDefault, {
+                class: 'right_column_list_elems',
+              });
+
+              const link = createAndAppend('a', rightColumnListElems, {
+                href: contributor.html_url,
+                target: '_blank',
+              });
+              const rightColumnContProps = createAndAppend('div', link, {
+                class: 'right_column_cont_props',
+              });
+              createAndAppend('img', rightColumnContProps, {
+                class: 'right_column_img',
+                src: contributor.avatar_url,
+              });
+
+              createAndAppend('p', rightColumnContProps, {
+                class: 'contName',
+                text: contributor.login,
+              });
+
+              createAndAppend('p', rightColumnContProps, {
+                class: 'badge',
+                text: contributor.contributions,
+              });
+            });
+          });
+        }
+
+        createRightPanel();
+
+        select.addEventListener('change', () => {
+          remove();
```

Comment:
You can also put the `remove()` inside the listener you are having below.

Reply:
Absolutely. I failed to noticed that as well.